src/colorizer/utils/dataset_load_utils.ts

```diff
@@ -0,0 +1,132 @@
+import { LocationState } from "../../types";
+import { ReportErrorCallback, ReportLoadProgressCallback, ReportWarningCallback } from "../types";
+import { isUrl, UrlParam } from "./url_utils";
+
+import Collection, { CollectionLoadOptions, DatasetLoadOptions } from "../Collection";
+import Dataset from "../Dataset";
+import { IArrayLoader } from "../loaders/ILoader";
+
+export const enum LoadResultType {
+  SUCCESS,
+  LOAD_ERROR,
+  MISSING_DATASET,
+}
+
+type LoadResult<T> =
+  | { type: LoadResultType.SUCCESS; value: T }
+  | { type: LoadResultType.LOAD_ERROR; message: string }
+  | { type: LoadResultType.MISSING_DATASET };
```

Comment:
The missing dataset currently triggers a different style of error from our usual (bright red) load errors, so I capture it as a different result. It has different actions ("Return to homepage") too.

![image](https://github.com/user-attachments/assets/460cb3db-7329-4337-bb6f-113b8265082c)